src/js/routes/Ballot/Ballot.jsx

```diff
@@ -97,11 +97,11 @@ export default class Ballot extends Component {
     let ballot = this.state.ballot;
     let ballot_props = BallotStore.ballot_properties;
     let ballot_length = 0;
-    if (ballot !== null && ballot.length !== null) {
+    if (typeof ballot !== "undefined" && ballot !== null && ballot.length !== null) {
```

Comment:
Or better yet, return the loading wheel if you don't have a ballot. Then you don't have to check any more.